src/x11_monitor.c

```diff
@@ -114,7 +114,7 @@ GLFWbool _glfwSetVideoModeX11(_GLFWmonitor* monitor, const GLFWvidmode* desired)
         if (_glfwCompareVideoModes(&current, best) == 0)
             return GLFW_TRUE;
 
-        sr = XRRGetScreenResources(_glfw.x11.display, _glfw.x11.root);
+        sr = _glfwXRRGetScreenResources();
```

Comment:
This invalid call emits no warnings because of the the bad declaration below. It will pull its arguments from random garbage on the stack and likely segfault as the first parameter is a pointer.